python/ray/air/_internal/remote_storage.py

```diff
@@ -11,6 +11,7 @@
 from typing import Any, Dict, List, Optional, Tuple
 
 from ray.air._internal.filelock import TempFileLock
+from ray.air._internal.uri_utils import is_uri
```

Comment:
Code organization wise, I think that we should aim to remove this `remote_storage.py` file fully in 2.8. To make this easier, how about copying/putting any FS utils inside the `StorageContext` class instead of depending on this file?

I.e., let's make the new code path never depend on `remote_storage.py`. That way, we don't have old/new code entangled with feature flags here.

Reply:
@ericl Here's the list of helpers that I want to keep from `remote_storage`, since I'd need to re-implement a lot of this functionality anyways if they were removed. I'll move these out and make them require a `fs` argument (removing the legacy path parsing).
* `upload_to_uri`
* `download_from_uri`
* `list_at_uri`
* `delete_at_uri` (this basically just wraps `fs.delete_dir` in a try/catch)
* `is_directory`

Reply:
Sounds good. Btw, one thought was maybe we can make these private methods in storage, and add helpers so that Tune/Train code can have more high level operations like `storage.upload_to_current_checkpoint_dir(local_dir)`.

This would just be for readability.

Reply:
Got it, can try this out